mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp

```diff
@@ -446,6 +471,72 @@ void CHudMapFinishedDialog::SetCurrentPage(int pageNum)
     m_pZoneVelMax->SetText(CConstructLocalizedString(m_pwVelMax, m_pRunStats ? m_pRunStats->GetZoneVelocityMax(m_iCurrentPage, velType) : 0.0f));
 }
 
+bool CHudMapFinishedDialog::LoadPlayerBestTime()
+{
+    const auto pPlayer = C_MomentumPlayer::GetLocalMomPlayer();
+    if (!pPlayer)
+    {
+        return false;
+    }
+    const auto pRunData = pPlayer->GetCurrentUIEntData();
+
+    CMomReplayBase *pBase = MomUtil::GetBestTime(g_pGameRules->MapName(), pRunData->m_flTickRate,
+                                                 pRunData->m_iCurrentTrack, pRunData->m_iRunFlags);
+
+    if (!pBase)
+    {
+        return false;
+    }
+    
+    m_pPbRunStats.FullyCopyFrom(*pBase->GetRunStats());
+    m_flPbOverallTime = pBase->GetRunTime();
```

Comment:
I was going to say you could probably just use the stats of the run but then I saw that this line was commented out:
https://github.com/momentum-mod/game/blob/develop/mp/src/game/server/momentum/mom_replay_system.cpp#L301

I'm actually not sure why this is. It's valuable to store the total ticks taken to do the map in the 0th zone. Perhaps try implementing this line in this branch, and see if runs still submit for you?